hudi-client/src/main/java/org/apache/hudi/client/HoodieCleanClient.java

```diff
@@ -85,7 +85,11 @@ public HoodieCleanMetadata clean(String startCleanTime) throws HoodieIOException
     // If there are inflight(failed) or previously requested clean operation, first perform them
     table.getCleanTimeline().filterInflightsAndRequested().getInstants().forEach(hoodieInstant -> {
       LOG.info("There were previously unfinished cleaner operations. Finishing Instant=" + hoodieInstant);
-      runClean(table, hoodieInstant);
+      try {
+        runClean(table, hoodieInstant);
+      } catch (Exception e) {
+        LOG.warn("Failed to perform previous clean operation, instant: " + hoodieInstant, e);
```

Comment:
@lamber-ken : Can we rollback the corrupted instant seamlessly in this case ?

Can you also add a CLI (called timeline upgrade) which would read the timeline  and fix upgrade issues.  For now, the CLI can just handle only known issue - to deal with corrupted inflight clean instants.  We can have users run them before upgrade for now and also see if this can be done seamlessly.

Reply:
Good idea 👍 , will update the pr.

Reply:
Sorry for delay, had updated the pr, any suggestion are welcome, thanks @bvaradar